### PR TITLE
Multiple objects of the same type can be added now. Fixed #51

### LINKED ISSUE #51: Level Builder - Unable to generate two sprites of the same type
Unless the sprite is renamed in the hierarchy, we can't import the same sprite for the second time right now.

### PR DESCRIPTION
**Pull request by OlafSzmidt**

### CODE REVIEW

File: UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs [hunk 1]
```diff
@@ -13,6 +13,10 @@
 
 class GeneratorMenu : IMenu
 {
+	// Autoincrementing value used for generating obstacles (so we don't have the same
+	// filenames twice as Unity does not allow for this.
```
Comment (CelineBoudier) [inline UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs]: missing closing parenthesis 🙀

File: UnityProject/Assembly-CSharp-Editor.csproj [hunk 2]
```diff
@@ -22,7 +22,7 @@
...
...RENDERER_FLIPPING;ENABLE_SPRITES;ENABLE_TERRAIN;ENABLE_DIRECTOR;ENABLE_UNET;ENABLE_UNITYEVENTS;ENABLE_WEBCAM;ENABLE_WWW;ENABLE_CLOUD_SERVICES_COLLAB;ENABLE_CLOUD_SERVICES_COLLAB_SOFTLOCKS;ENABLE_CLOUD_HUB;ENABLE_CLOUD_PROJECT_ID;ENABLE_CLOUD_SERVICES_USE_WEBREQUEST;ENABLE_CLOUD_SERVICES_UNET;ENABLE_CLOUD_SERVICES_BUILD;ENABLE_CLOUD_LICENSE;ENABLE_WEBSOCKET_CLIENT;ENABLE_DIRECTOR_AUDIO;ENABLE_TIMELINE;ENABLE_EDITOR_METRICS;ENABLE_EDITOR_METRICS_CACHING;ENABLE_NATIVE_ARRAY;ENABLE_SPRITE_MASKING;INCLUDE_DYNAMIC_GI;INCLUDE_GI;ENABLE_MONO_BDWGC;RENDER_SOFTWARE_CURSOR;INCLUDE_PUBNUB;ENABLE_PLAYMODE_TESTS_RUNNER;ENABLE_VIDEO;ENABLE_RMGUI;ENABLE_CUSTOM_RENDER_TEXTURE;ENABLE_STYLE_SHEETS;PLATFORM_WEBGL;UNITY_WEBGL;UNITY_WEBGL_API;UNITY_DISABLE_WEB_VERIFICATION;UNITY_GFX_USE_PLATFORM_VSYNC;ENABLE_CRUNCH_TEXTURE_COMPRESSION;ENABLE_UNITYWEBREQUEST;ENABLE_CLOUD_SERVICES;ENABLE_CLOUD_SERVICES_ADS;ENABLE_CLOUD_SERVICES_ANALYTICS;ENABLE_CLOUD_SERVICES_PURCHASING;ENABLE_CLOUD_SERVICES_CRASH_REPORTING;ENABLE_VR;ENABLE_IL2CPP;NET_2_0_SUBSET;DEVELOPMENT_BUILD;ENABLE_PROFILER;UNITY_ASSERTIONS;UNITY_EDITOR;UNITY_EDITOR_64;UNITY_EDITOR_LINUX;ENABLE_NATIVE_ARRAY_CHECKS;UNITY_TEAM_LICENSE</DefineConstants>
```
Comment (CelineBoudier) [inline UnityProject/Assembly-CSharp-Editor.csproj]: ?

Comment (OlafSzmidt) [reply UnityProject/Assembly-CSharp-Editor.csproj]: This is down to personal preferance whether we need this or not. Afaik, .csproj keeps the structure of the files in the code, so it's needed for our project - but MonoDevelop is able to create these every time it runs the code.

We had some issues with the auto creation so we just include local .csproj with every change we make. We can go back to excluding them if you wish @CelineBoudier.

Comment (CelineBoudier) [reply UnityProject/Assembly-CSharp-Editor.csproj]: If we had issues I'd rather we keep them, but it's ugly ;)

File: UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs [hunk 3]
```diff
@@ -57,40 +61,40 @@ private void GeneratorGUI(Type generatorType, string buttonName, GenerationData
...
-			// We use the Builder pattern so that the GUI logic can get updated easily 
+			// We use the Builder pattern so that the GUI logic can get updated easily
 			builder = builder.ByCoord(0.0f, 0.0f);
-			if (data.idx != 0) 
+			if (data.idx != 0)
 				builder = builder.ByPath (sprites [data.idx]);
-		
-			if (data.width != "") 
+
+			if (data.width != "")
```
Comment (CelineBoudier) [inline UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs]: how come width and heights are strings?

Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs]: I looked for a while now, in both `GeneratorMenu` and `SpriteGeneratorBuilder` and can't find an answer for this. Maybe @danalex97?

Comment (danalex97) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs]: I think I kept using [text fields](https://docs.unity3d.com/ScriptReference/EditorGUILayout.TextField.html) while prototyping and forgot to replace those with [float fields](https://docs.unity3d.com/ScriptReference/EditorGUI.FloatField.html) or [int fields](https://docs.unity3d.com/ScriptReference/EditorGUI.IntField.html). We can open an issue to making all the GUI elements more consistent.

Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs]: Issue is open. @CelineBoudier - do you want me to fix this now for this PR or do it later on when I go through the issues?

Comment (CelineBoudier) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/Menus/GeneratorMenu.cs]: It's ok to look at it later as technically this works I guess

**Pull request merged**